### PR TITLE
deleteFlashcard

### PR DESCRIPTION
**Pull request by DorotaHeczko**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
The task goal is to delete an existing flashcard from the database when deleting one.

### Requirements
1. Fetch API must be used
2. Following endpoint must be used:
```
curl -X DELETE \
https://training.nerdbord.io/api/v1/fischkapp/flashcards/:id \
-H 'Authorization: secret_token' \
```

### Code Changes Review

The code changes provided are in line with the task requirements. The Fetch API is used to delete a flashcard from the database using the DELETE method and the specified endpoint. The 'Authorization' header is correctly set to 'secret_token'. 

Here is the relevant code:

```javascript
async function deleteFlashcard(id: string): Promise<void> {
  const response = await fetch(
    `https://training.nerdbord.io/api/v1/fischkapp/flashcards/${id}`,
    {
      method: "DELETE",
      headers: {
        Authorization: "secret_token",
        "Content-Type": "application/json",
      },
    }
  );

  if (!response.ok) {
    const responseData = await response.json();
    throw new Error(responseData.message || "Failed to delete the flashcard.");
  }
}
```

The `onDeleteFunction` function has been updated to call `deleteFlashcard` function and handle any errors that might occur during the deletion process. This is a good practice as it ensures that any errors that occur during the deletion process are caught and handled appropriately.

```javascript
const onDeleteFunction = async (id: string) => {
  try {
    await deleteFlashcard(id);
    setCards((prevCards) => prevCards.filter((card) => card.id !== id));
  } catch (error) {
    console.error("Error while deleting the flashcard:", error);
  }
};
```

In conclusion, the code changes cover the task requirements and there are no apparent bugs in the changed code.
